### PR TITLE
File ACLs migration

### PR DESCRIPTION
**Pull request by bion**
Adds a route and background task for adding ACLs to existing StoredFiles. A logged in Seattle admin (does not need any specific admin role, so any Seattle staff member) can trigger the data migration by visiting `/admin/migrateStoredFileAcls` in their browser. The server will log messages that include the strings `StoredFileMigrationCompleted` and `StoredFileMigrationError` that technical staff can use to assess the success of the migration.

For context on StoredFileAcls: https://github.com/seattle-uat/civiform/pull/2689

fixes https://github.com/seattle-uat/civiform/issues/2709

### CODE REVIEW

**Review by swatkat1 — APPROVED** [review]

File: server/app/tasks/StoredFileAclMigrationTask.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import services.program.ProgramNotFoundException;
+import services.program.ProgramService;
+
+/**
+ * Migrates {@link StoredFile} to ensure they have correct {@link auth.StoredFileAcls} by looping
+ * through all submitted applications, inspecting their files, and adding ACLs if they are missing
+ * for the program the application is associated with. This task is idempotent and can be run
+ * multiple times.
+ *
+ * <p>This can be deleted after the data in Seattle production has been migrated.
```
Comment (gwendolyngoetz) [inline server/app/tasks/StoredFileAclMigrationTask.java:28]: Oh this is just a one time thing to fix existing data?

File: server/app/controllers/admin/DataMigrationsController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  @Secure(authorizers = Authorizers.Labels.ANY_ADMIN)
+  public Result migrateStoredFileAcls() {
+    actorSystem
+        .scheduler()
+        .scheduleOnce(
+            Duration.ofMillis(10),
+            () -> storedFileAclMigrationTaskProvider.get().run(),
+            executionContext);
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/controllers/admin/DataMigrationsController.java:50]: to clarify terminology, an [action](https://www.playframework.com/documentation/2.8.x/JavaActions) is a Play HTTP handler. a *task* is piece of code that is meant to be executed out-of-band from a normal HTTP request-response cycle. here we have an action that uses Play's background scheduler (the Akka scheduler) to schedule a task to run in the background 10 milliseconds after the request. calling the action once, by sending an http request to its route, will schedule the task once. calling it multiple times (multiple people calling it, opening it multiple tabs, refreshing the browser page after the initial request) will schedule the task multiple times.

if multiple instances of the task are running at the same time it shouldn't be a big deal. all of the ACL updates are done in serializable database transactions, and the code handles the case where a file's ACLs have already been updated by not doing anything for the file.

File: server/app/controllers/admin/DataMigrationsController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import javax.inject.Inject;
+import javax.inject.Provider;
+import org.pac4j.play.java.Secure;
+import play.mvc.Result;
+import scala.concurrent.ExecutionContext;
+import tasks.StoredFileAclMigrationTask;
+
+/**
+ * Controller enabling admins to trigger data migrations in production. Need not serve a UI. all
+ * actions must be protected by @Secure(authorizers = Authorizers.Labels.ANY_ADMIN). Actions should
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/controllers/admin/DataMigrationsController.java:20]: I've tried that in the past and it hasn't worked.

**Review by gwendolyngoetz — APPROVED** [review]

File: server/app/tasks/StoredFileAclMigrationTask.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    this.applicationRepository = checkNotNull(applicationRepository);
+    this.baseUrl = checkNotNull(configuration).getString("base_url");
+    this.database = DB.getDefault();
+    this.programService = checkNotNull(programService);
+    this.storedFileRepository = checkNotNull(storedFileRepository);
+  }
+
+  @Override
+  public void run() {
+    // Using a counter class here because Java does not allow lambdas to mutate variables
```
Comment (shanemc-goog) [inline server/app/tasks/StoredFileAclMigrationTask.java:54]: FYI The typical approach is to use a AtomicInteger / Long

**Pull request merged**